### PR TITLE
 Fix JaxRS-Jersey template for files upload

### LINKED ISSUE #2474: [BUG] [Java | JaxRS-Jersey] Bad file upload code generation
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?

##### Description

The return generated for a POST method that takes a file as input and file as output don't match the generated class method. Has two invalid arguments (bodyInputStream and bodyDetail) instead of body

```
public Response getApplication(@ApiParam(value = "input APK" ,required=true) @Valid File body,@Context SecurityContext securityContext) throws NotFoundException {
        return delegate.getApplication(bodyInputStream, bodyDetail,securityContext);
    }
````

##### openapi-generator version

4.0.0-beta2

##### OpenAPI declaration file content or url

```yaml
/getApplication:
    post:
      operationId: getApplication
      summary: getting application
      tags: ['Android']
      requestBody:      
        description: input APK
        required: true
        content:
          application/vnd.android.package-archive:
            schema:
              type: string
              format: binary
      responses:
        '200':
          description: output APK
          content:
            application/vnd.android.package-archive:
              schema:
                type: string
                format: binary
```

##### Command line used for generation
```
openapi-generator generate -i .\api-specs.yaml -g jaxrs-jersey -o ./tmp/test/
```
##### Output
```
@POST
    
    @Consumes({ "application/vnd.android.package-archive" })
    @Produces({ "application/vnd.android.package-archive" })
    @io.swagger.annotations.ApiOperation(value = "getting application", notes = "", response = File.class, authorizations = {
        @io.swagger.annotations.Authorization(value = "ApiKeyAuth")
    }, tags={ "Android", })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "output APK", response = File.class) })
    public Response getApplication(@ApiParam(value = "input APK" ,required=true) @Valid File body
,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.getApplication(bodyInputStream, bodyDetail,securityContext);
    }
```

##### Expected Output
```
@POST
    
    @Consumes({ "application/vnd.android.package-archive" })
    @Produces({ "application/vnd.android.package-archive" })
    @io.swagger.annotations.ApiOperation(value = "getting application", notes = "", response = File.class, authorizations = {
        @io.swagger.annotations.Authorization(value = "ApiKeyAuth")
    }, tags={ "Android", })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "output APK", response = File.class) })
    public Response getApplication(@ApiParam(value = "input APK" ,required=true) @Valid File body
,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.getApplication(body ,securityContext);
    }
```

### PR DESCRIPTION
**Pull request by Zomzog**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Remove specific File management from template to align it with the service. 

fix #2474

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03)

### CODE REVIEW

**Comment by karismann** [comment]
thanks for you PR but with this change, it seem like there is a regression with type file in OAS2 :

```
      parameters:
        - in: formData
          name: jarFile
          description: Jar File to upload
          required: true
          type: file
```


```
    public Response addPet(
            @FormDataParam("jarFile") InputStream jarFileInputStream,
            @FormDataParam("jarFile") FormDataContentDisposition jarFileDetail
,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.addPet(jarFile, securityContext);
    }
```

you can refer to https://github.com/swagger-api/swagger-codegen/issues/3300

File: modules/openapi-generator/src/main/resources/JavaJaxRS/api.mustache [hunk 1]
```diff
@@ -76,7 +76,7 @@ public class {{classname}}  {
         {{/hasMore}}{{/responses}} })
     public Response {{nickname}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}},{{/allParams}}@Context SecurityContext securityContext)
     throws NotFoundException {
-        return delegate.{{nickname}}({{#allParams}}{{#isFile}}{{paramName}}InputStream, {{paramName}}Detail{{/isFile}}{{^isFile}}{{paramName}}{{/isFile}},{{/allParams}}securityContext);
+        return delegate.{{nickname}}({{#allParams}}{{paramName}}, {{/allParams}}securityContext);
```
Comment (karismann) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/api.mustache]: actually the real problem is the missing `isFormParam `in this condition. the formParams.mustache who generate an InputStream and an FormDataContentDisposition variables are conditionned with the `isFormParam` so this line should be fixed with :  
```
return delegate.{{nickname}}({{#allParams}}{{#isFormParam}}{{#isFile}}{{paramName}}InputStream, {{paramName}}Detail{{/isFile}}{{/isFormParam}}{{^isFile}}{{paramName}}{{/isFile}},{{/allParams}}securityContext);
```
to solve the issue

Comment (Zomzog) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/api.mustache]: I've changed the template with something inspired by your solution.

I've tried it with the issue's sample and this one :

`
  /Plugin:
    post:
      summary: uploads an image
      description: ''
      operationId: uploadFile
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                additionalMetadata:
                  description: Additional data to pass to server
                  type: string
                file:
                  description: file to upload
                  type: string
                  format: binary
`

Comment (karismann) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/api.mustache]: thanks . can you please push your new commit to your branch "2474" to update the PR ?

Comment (Zomzog) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/api.mustache]: It can be useful. Done.

**Comment by wing328** [comment]
Please update the samples:
```
     @GET
     @Path("/logout")
@@ -161,6 +161,6 @@ public class UserApi  {
 ,@ApiParam(value = "Updated user object" ,required=true) @NotNull @Valid User user
 ,@Context SecurityContext securityContext)
     throws NotFoundException {
-        return delegate.updateUser(username,user,securityContext);
+        return delegate.updateUser(username, user, securityContext);
     }
 }
Perform git status
On branch Zomzog-2474
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/AnotherFakeApi.java
	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/FakeApi.java
	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/FakeClassnameTestApi.java
	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/PetApi.java
	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/StoreApi.java
	modified:   samples/server/petstore/jaxrs-jersey/src/gen/java/org/openapitools/api/UserApi.java
```

Tested via https://circleci.com/gh/OpenAPITools/openapi-generator/5930#tests/containers/2

**Comment by tduperron** [comment]
I've updated java-jaxrs-petstore-server-all.sh to run also this sample and run it.

**Comment by Zomzog** [comment]
I've updated java-jaxrs-petstore-server-all.sh to run also this sample and run it.

**Comment by wing328** [comment]
CircleCI test passed via https://circleci.com/gh/OpenAPITools/openapi-generator/5932

**Pull request merged**